src/handlers/login-handler.ts

```diff
@@ -15,15 +15,15 @@ export class LoginHandler extends HttpHandler {
     console.log("LoginHandler::constructor");
   }
 
-  async handleAsync (context: HttpSolidContext): Promise<HttpHandlerResponse> {
+  async handleAsync (context: AuthnContext): Promise<HttpHandlerResponse> {
     if (!context.authn) {
       return {
         status: 401,
         headers: {},
       };
     }
 
-    const idp = context.request.body?.idp;
+    const idp = JSON.parse(context.request.body)['idp'];
```

Comment:
If the request sets proper `Content-Type: application/json` we are getting body parsed (assuming handlersjs) does it.

Since we want to handle the case where it wasn't set we need to make sure `request.body` exists in the first place.

Probably we should check the content type of the response and if different than `application/json` we should respond with 400 Invalid request right away.

Reply:
@JasperVandenberghen @AbelVandenBriel

Reply:
No I don't believe there is. We usually just preform a simple check in the handle and throw this one: packages/handlersjs-http/lib/errors/bad-request-http-error.ts